src/utils/string/index.ts

```diff
@@ -0,0 +1,23 @@
+export const msToDuration = (ms: number): string => {
+  const days = Math.floor(ms / (24 * 60 * 60 * 1000));
+  const daysMilli = ms % (24 * 60 * 60 * 1000);
+  const hours = Math.floor(daysMilli / (60 * 60 * 1000));
+  const hoursMilli = ms % (60 * 60 * 1000);
+  const minutes = Math.floor(hoursMilli / (60 * 1000));
+  const minutesMilli = ms % (60 * 1000);
+  const seconds = Math.floor(minutesMilli / 1000);
+  const milliseconds = ms % 1000;
```

Comment:
The math here implies ms is in microseconds, not milliseconds

Reply:
there are a thousand milliseconds in a second, so I see how many milliseconds are leftover after we account for seconds. I think this works.

Reply:
You're using the wrong variable then - ms is the input (which I'm assuming to be milliseconds). I'm looking at a test which took 25 secs, but shows as 25ms in this table

Reply:
it looks like I'm using seconds as ms. I made a code change in spruce to convert seconds to ms before giving the number to `msToDuration`